packages/nuxt/src/app/components/nuxt-link.ts

```diff
@@ -280,8 +279,15 @@ export function defineNuxtLink (options: NuxtLinkOptions) {
 
         // Resolves `to` value if it's a route location object
         // converts `""` to `null` to prevent the attribute from being added as empty (`href=""`)
-        const href = typeof to.value === 'object' ? router.resolve(to.value)?.href ?? null : to.value || null
+        let href = typeof to.value === 'object' ? router.resolve(to.value)?.href ?? null : to.value || null
 
+        // joins with `baseURL` if it's an relative link
+        if (import.meta.client && href && href.startsWith('/') && !href.startsWith('//')) {
```

Comment:
Have you tried to test it by `pnpm vitest nuxt-link`?

In test cases, it will report an error for lack of context.

May be we should fix it by:

```suggestion
        if (hasInjectionContext() && href && href.startsWith('/') && !href.startsWith('//')) {
```

Reply:
Have you ever attempted to include this in tests?

```
// nuxt-link.test.ts
// Mocks Nuxt `useRuntimeConfig()`
vi.mock('../src/app/nuxt', () => ({
  useRuntimeConfig: () => ({
    app: {
      baseURL: '/'
    }
  })
}))
```

Reply:
I see, thank you for pointing out the correct approach!